### PR TITLE
Fix inode check for smb -> file

### PR DESCRIPTION
**Pull request by shyaler**
1. Save file id in StoredObject.inode
3. Add isRootDirectory property to sync comperator -> to avoid inode validation
4. Add debug instructions
5. Add manual testing instructions

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    // Use IntelliSense to learn about possible attributes.
```
Comment (aymanjarrousms) [inline .vscode/launch.json:2]: Recommend to drop this

File: cmd/syncComparator.go [hunk 2]
```diff
@@ -453,7 +453,8 @@ func (f *syncDestinationComparator) processIfNecessary(destinationObject StoredO
 			//
 
 			// Inode should be set for both source and destination.
-			if destinationObject.inode == 0 || sourceObjectInMap.inode == 0 {
+			// Root directory doesnt have an ID
+			if !destinationObject.isRootDirectory && (destinationObject.inode == 0 || sourceObjectInMap.inode == 0) {
```
Comment (aymanjarrousms) [inline cmd/syncComparator.go:457]: Is this for checking only the root directory?

File: cmd/zc_traverser_file.go [hunk 3]
```diff
@@ -112,6 +119,7 @@ func (t *fileTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 
 		// We need to omit some properties if we don't get properties
 		lmt := time.Time{}
+		fileId := ""
```
Comment (aymanjarrousms) [inline cmd/zc_traverser_file.go:122]: var fileId string, be default it's empty string and not null.
I think that it's preferable by Go

**Review by aymanjarrousms — APPROVED** [review]

File: cmd/syncComparator.go (hunk 2 above)
Comment (shyaler) [reply cmd/syncComparator.go:457]: no, its for excluding root directory from the inode check

**Review by aymanjarrousms — APPROVED** [review]

**Review by OmarKhatib158 — APPROVED** [review]

**Pull request merged**